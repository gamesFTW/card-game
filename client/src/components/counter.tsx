import * as React from 'react';

interface Props {
}
interface State {
  count: number;
}

export class Counter extends React.Component<Props, State> {
  interval: number;
  state = { count: 0 };

  componentWillMount() {
    const incrementCounter = () => {
      this.setState({ count: this.state.count + 1 });
    };
    this.interval = setInterval(incrementCounter, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    return (
      <div>
        <div>Counter: {this.state.count}</div>
      </div>
    );
  }
}

export default Counter;
