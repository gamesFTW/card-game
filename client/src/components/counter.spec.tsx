import * as React from 'react';
import { shallow } from 'enzyme';
import { Counter } from './counter';

describe('Counter', () => {
  const component = shallow(<Counter />);

  it('should match a snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should initialize state', () => {
    expect(component.state()).toEqual({ count: 0 });
  });

  it('should change state after a second', (done) => {
    setTimeout(() => {
      expect(component.state()).not.toEqual({ count: 0 });
      done();
    }, 1500);
  });
});
