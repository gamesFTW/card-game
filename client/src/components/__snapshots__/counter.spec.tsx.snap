// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Counter: 
        0
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Counter: ",
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Counter: ",
        0,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Counter: 
          0
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Counter: ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Counter: ",
          0,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
